### ISSUE TITLE
Assignment 1 part A

### ISSUE THREAD

**Issue by adarshdas1995** [goodboyanush/iiit-bangalore-march-april-2019#3]
Hi Anush,
I went through the keras implementation of the Upsampling layer. It simply says that it repeats the rows and columns values.
Here is the link by keras team.
https://github.com/keras-team/keras/blob/8ed57c168f171de7420e9a96f9e305b8236757df/keras/layers/convolutional.py#L1552
Although I implemented the same thing, it still does not give me the right output. 
Any suggestions ?
![input](https://user-images.githubusercontent.com/42201757/56040900-1c4e2f00-5d55-11e9-8b51-e02a0b896684.png)
![output](https://user-images.githubusercontent.com/42201757/56040901-1c4e2f00-5d55-11e9-9b7f-a2346dde6009.png)

**Comment by satu0king** [comment]
Have you used sigmoid and relu activations?

**Comment by adarshdas1995** [comment]
Oop's ! I forgot about the activations. Will implement them.
Edit : I tried it. Normal thresholding with numpy to do relu activations results in a binary image. Another method yielded results similar to the one showcased in the issue. Any other suggestions?

**Comment by satu0king** [comment]
> Edit : I tried it. Normal thresholding with numpy to do relu activations results in a binary image. Another method yielded results similar to the one showcased in the issue. Any other suggestions?

I just did `np.maximum(0,X)` to do relu. For sigmoid, I used 

```
def sigmoid(x):
    return 1./(1.+ np.exp(-x))
sigmoid = np.vectorize(sigmoid)
```

Maybe you can show your current images.

**Comment by progvasu** [comment]
@adarshdas1995 you could test your implementations for conv, pool and upsampling layers individually comparing them with keras output. It helps to narrow down the bugs.

**Comment by goodboyanush** [comment]
@adarshdas1995 Upsampling2D by default calls `K. resize_images` function. 

https://github.com/keras-team/keras/blob/2fadce57f0e12cf96f11fa1d42286986eccd70a8/keras/backend/tensorflow_backend.py#L2194

It does nearest neighbour resizing
